src/commands/createManagedEnvironment/createManagedEnvironment.ts

```diff
@@ -3,17 +3,16 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { IActionContext, ICreateChildImplContext } from "@microsoft/vscode-azext-utils";
+import { SubscriptionTreeItemBase } from "@microsoft/vscode-azext-azureutils";
+import { IActionContext } from "@microsoft/vscode-azext-utils";
 import { ext } from "../../extensionVariables";
 import { ManagedEnvironmentTreeItem } from "../../tree/ManagedEnvironmentTreeItem";
 import { SubscriptionTreeItem } from "../../tree/SubscriptionTreeItem";
 import { IManagedEnvironmentContext } from "./IManagedEnvironmentContext";
 
-export async function createManagedEnvironment(context: IActionContext & Partial<ICreateChildImplContext> & Partial<IManagedEnvironmentContext>, node?: SubscriptionTreeItem): Promise<ManagedEnvironmentTreeItem> {
-    if (!node) {
-        node = await ext.tree.showTreeItemPicker<SubscriptionTreeItem>(SubscriptionTreeItem.contextValue, context);
-    }
-
-    const keNode: ManagedEnvironmentTreeItem = await node.createChild(context);
+export async function createManagedEnvironment(context: IActionContext, node?: SubscriptionTreeItemBase): Promise<ManagedEnvironmentTreeItem> {
+    node = await ext.rgApi.appResourceTree.showTreeItemPicker<SubscriptionTreeItemBase>(SubscriptionTreeItemBase.contextValue, context);
```

Comment:
So I wasn't sure how best to handle this, but basically my thought process was something like this.  If we use the if (!node) method, then the flow would be something like:

1. If you right click on the node to initiate a create, node will enter command as `GroupTreeItemBase`
2. If you initiate a create via QP, node will enter command as `undefined` and then turn into `SubscriptionTreeItemBase`

The command will work in both cases because the information it needs is on the treeItem; however, I thought it made more sense to have it so that the node was always a type that we could consistently expect - hence no if statement and just always re-initialize node to `SubscriptionTreeItemBase` in all cases.

Let me know if that makes sense or if I'm thinking about this in a weird way.  I'm thinking maybe it was a bad idea to label the incoming node type as `SubscriptionTreeItemBase` - maybe I can just say something like `_node?: AzExtTreeItem` and just initialize the subscription node reference as a separate variable to avoid any potential confusion down the road

Reply:
The desired behavior is to avoid prompting for the subscription if at all possible. If the command has been run from the Container Apps type group tree item context menu, we shouldn't need to prompt for the subscription.

Although it's a `GroupTreeItemBase`, we can just cast it as a `SubscriptionTreeItem` since it has the `.subscription` property. It's not ideal, but we do that in every create command IIRC. [Checkout the Create Function App command as an example](https://github.com/microsoft/vscode-azurefunctions/blob/alex/mainjs/src/commands/createFunctionApp/createFunctionApp.ts#L13)